src/Common/Pagination/PageSizeSelector.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useRef, useState } from 'react'
+import { PageSizeItemsProps, PageSizeOption, PageSizeSelectorProps } from './types'
+import { useClickOutside } from '../Hooks'
+import { getDefaultPageValueOptions } from './utils'
+
+const PageSizeItems = ({
+    optionValue,
+    options,
+    setOptions,
+    handleCloseDropdown,
+    changePageSize,
+}: PageSizeItemsProps) => {
+    const handlePageSizeSelect = () => {
+        const newOptions = options.map((_option) => ({
+            value: _option.value,
+            selected: _option.value === optionValue,
+        }))
+        setOptions([...newOptions])
+        handleCloseDropdown()
+        changePageSize(optionValue)
+    }
+
+    return (
+        <div key={optionValue} className="select__item" onClick={handlePageSizeSelect}>
+            {optionValue}
+        </div>
+    )
+}
+
+const PageSizeSelector = ({ pageSizeOptions, pageSize, changePageSize }: PageSizeSelectorProps) => {
+    const defaultOptions = getDefaultPageValueOptions(pageSizeOptions).map((option) => ({
+        value: option.value,
+        selected: option.value === pageSize,
+    }))
+
+    const [isDropdownOpen, setIsDropdownOpen] = useState<boolean>(false)
+    const [options, setOptions] = useState<PageSizeOption[]>(defaultOptions)
+
+    const handleCloseDropdown = () => {
+        setIsDropdownOpen(false)
+    }
+
+    const handleDropdownToggle = () => {
+        setIsDropdownOpen(!isDropdownOpen)
+    }
+
+    const dropdownRef = useRef<HTMLDivElement>(null)
+    useClickOutside(dropdownRef, handleCloseDropdown)
+
+    const selectedOption = options.find((option) => option.selected)
+    const font = isDropdownOpen ? 'fa fa-caret-up' : 'fa fa-caret-down'
+    return (
+        <div ref={dropdownRef}>
```

Comment:
same: use react select

Reply:
imported the logic from dashboard, even have left a note stating normalisation might benefit here.